### PR TITLE
Change logout endpoint to POST request

### LINKED ISSUE #97: Change the logout endpoint to a POST request instead of a GET request
As it is now, it's possible for someone to log someone else out by linking them to the logout page (or linking to an image with source "porybox.com/logout").

### PR DESCRIPTION
**Pull request by not-an-aardvark**
Fixes #97

### CODE REVIEW

File: api/controllers/AuthController.js [hunk 1]
```diff
@@ -68,12 +68,8 @@ module.exports = {
    * @param {Object} res
    */
   logout: function (req, res) {
-    req.logout();
-
-    // mark the user as logged out for auth purposes
-    req.session.authenticated = false;
-
-    res.redirect('/');
+    req.session.destroy();
```
Comment (YaManicKill) [inline api/controllers/AuthController.js]: What's the difference here with what it used to do? Is this purely because it's a POST request now?

Comment (not-an-aardvark) [inline api/controllers/AuthController.js]: Passport.js seems to behave a bit strangely with requests that come from websockets -- the req.logout function doesn't end up attached.

Also, using `res.redirect` doesn't work in general with websockets, so it returns a 200 instead and this tells the client to do a redirect.

Comment (YaManicKill) [inline api/controllers/AuthController.js]: Alright, I'm assuming that req.session.destroy() also implicitly sets authenticated to false then?

Comment (YaManicKill) [inline api/controllers/AuthController.js]: Tell you what, can you do a check here if we are websockets, and if we aren't, do a redirect. Just would make things easier if we do a Javascript-free frontend in the future.

Comment (YaManicKill) [inline api/controllers/AuthController.js]: Actually, nevermind. Let's leave that until we actually do a JS-free FE.

**Comment by YaManicKill** [comment]
:+1: merging

**Pull request merged**